File: src/bandcamp/bandcamp-objects.ts
export interface BandcampAlbumMetainfo {
	current: {
		title: string;
		type: string;
		release_date: string;
	},
	hasAudio: boolean;
	artist: string;
	trackinfo: BandcampTrackInfo[];
}

export interface BandcampTrackInfo {
	file: {
		"mp3-128": string;
	},
	title: string;
	track_num: number;
	has_lyrics: boolean;
}

File: src/parser/bandcamp_objects.rs
use serde::Deserialize;

#[derive(Deserialize, Debug)]
pub struct BandcampTralbum {
    pub trackinfo: Vec<BandcampTrack>,
    pub artist: String,
    pub current: BandcampCurrent,
    pub album_release_date: String,
}

#[derive(Deserialize, Debug)]
pub struct BandcampTrack {
    pub file: Option<BandcampFile>,
    pub title: String,
    pub track_num: u64,
}

#[derive(Deserialize, Debug)]
pub struct BandcampCurrent {
    pub title: String,
    pub r#type: String,
}

#[derive(Deserialize, Debug)]
pub struct BandcampFile {
    #[serde(rename = "mp3-128")]
    pub mp3: String,
}
